### ISSUE TITLE
Questions about extracting bf16-mixed consolidated weights from DeepSpeed checkpoints

### ISSUE THREAD

**Issue by blueColdSea** [bytedance/HLLM#16]
Hi,
I'm trying to continue trainning from checkpoints. The code is:
```
import lightning as L

fabric = L.Fabric()

checkpoint_file = os.path.join(checkpoint_dir, 'checkpoint/mp_rank_00_model_states.pt')
full_checkpoint = fabric.load(checkpoint_file)
```
However, it keeps reporting the following errors:
![image](https://github.com/user-attachments/assets/cc8f12a8-129b-41f3-8bd7-da9562d711a0)
Could you help me solve the problem?

**Comment by threestone965** [comment]
应该是需要在HLLM/code/路径下运行上面的代码

**Comment by blueColdSea** [comment]
好的，谢谢。但是我目前还有一个问题，现在还是没有办法中断训练的情况下，从checkpoint提取出优化器状态继续训练。代码中尽管提供了保存检查点的方法，但是并没有从检查点恢复训练的函数。因为这个项目使用了deepspeed和lighting框架，自定义了保存checkpoint的函数，让人很难适应。
我尝试了挺多方法进行从checkpoint提取出优化器状态继续训练的尝试，但是很遗憾，暂时没有尝试成功。
如果您能提供相对应的方法，或者提供一些思路，我想不胜感激！

**Comment by lixali** [comment]
Running under HLLM/code directory does not seem to be working for me. I make a soft link to REC folder inside the HLLM/model/HLLM-0.path/ directory and python code there.

**Comment by threestone965** [comment]
> 好的，谢谢。但是我目前还有一个问题，现在还是没有办法中断训练的情况下，从checkpoint提取出优化器状态继续训练。代码中尽管提供了保存检查点的方法，但是并没有从检查点恢复训练的函数。因为这个项目使用了deepspeed和lighting框架，自定义了保存checkpoint的函数，让人很难适应。 我尝试了挺多方法进行从checkpoint提取出优化器状态继续训练的尝试，但是很遗憾，暂时没有尝试成功。 如果您能提供相对应的方法，或者提供一些思路，我想不胜感激！


保存checkpoint的时候需要再加一个学习率调度器
    def _save_checkpoint(self, epoch, verbose=True):
        r"""Store the model parameters information and training information.

        Args:
            epoch (int): the current epoch id

        """
        state = {
            "model": self.model,
            "optimizer": self.optimizer,
            'scheduler': self.lr_scheduler,
            'config': self.config,
            'epoch': epoch,
            'cur_step': self.cur_step,
            'best_valid_score': self.best_valid_score,
            'rng_state': torch.get_rng_state(),
            'cuda_rng_state': torch.cuda.get_rng_state()
        }

resume时加载checkpoint

        if self.config['auto_resume']:
            checkpoint_files = [f for f in os.listdir(self.checkpoint_dir) if f.endswith('.pth')]
            if checkpoint_files:
                # Assuming the latest checkpoint is the one to resume from
                latest_checkpoint = max(checkpoint_files, key=lambda x: int(x.split('-')[1].split('.')[0]))
                checkpoint_path = os.path.join(self.checkpoint_dir, latest_checkpoint)
                self.logger.info(f"Loading checkpoint from {checkpoint_path}")
                state = {
                    "model": self.model,
                    "optimizer": self.optimizer,
                    'scheduler': self.lr_scheduler
                }

                remainder = self.lite.load(checkpoint_path, state)
                self.best_valid_score = remainder.pop('best_valid_score')
                self.cur_step = remainder.pop('cur_step')
                torch.set_rng_state(remainder.pop('rng_state'))
                torch.cuda.set_rng_state(remainder.pop('cuda_rng_state'))
                epoch = remainder.pop('epoch')
                self.start_epoch = epoch + 1

**Comment by blueColdSea** [comment]
真是太感谢了！
